### DISCUSSION TITLE
next-intl TypeError: path.startsWith is not a function when using createSharedPathnamesNavigation

### DISCUSSION THREAD

**Discussion by aboutjquery** [amannn/next-intl#1295] [Q&A]
I am starting with `example-app-router`.

Since I only need the same `pathnames` I modified

navigation.ts
```
import { createSharedPathnamesNavigation } from 'next-intl/navigation';
import { localePrefix, locales } from './config';

export const { Link, redirect, usePathname, useRouter } =
 createSharedPathnamesNavigation({ localePrefix, locales });
```

I did not import `pathnames`

When I use the language switcher in example, I get an error

```
TypeError: path.startsWith is not a function

Source
src\components\LocaleSwitcherSelect.tsx (36:14) @ replace

 34 | // are used in combination with a given `pathname`. Since the two will
 35 | // always match for the current route, we can skip runtime checks.
> 36 | router.replace({ pathname, params }, { locale: nextLocale });
 | ^
 37 | });
 38 | }
 39 |
```

Where did I go wrong? Please tell me, thank you

**Comment by lorenzpfei** [comment]
See: https://github.com/amannn/next-intl/issues/1240#issuecomment-2288033394.
This should fix your problem :)

**Reply by amannn** [reply]
Or you can upgrade to [`createNavigation`](https://next-intl-docs.vercel.app/blog/next-intl-3-22#create-navigation) which works the same way regardless of if you're using `pathnames` or not!
